src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -278,6 +324,89 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
                 syntax.Body, syntax.ExpressionBody, syntax, diagnostics);
         }
 
+        /// <summary>
+        /// Constructor only for auto-properties
+        /// </summary>
+        private SourcePropertyAccessorSymbol(
```

Comment:
>SourcePropertyAccessorSymbol [](start = 16, length = 28)

This constructor looks very similar to the previous one. Could we share code between the two?

Reply:
Or alternatively, could the two constructors be merged? Would callers and implementers of `CreateAccessorSymbol()` and `CreateExprBodiedAccessor()` benefit if there was one constructor and one combined `CreateAccessorSymbol()` method rather than two of each?

---
In reply to: [445044599](https://github.com/dotnet/roslyn/pull/45411#discussion_r445044599) [](ancestors = 445044599)

Reply:
There's already precedence for creating two constructors, one for ArrowExpression and one for AccessorDeclaration. I'd be fine with improving the sharing, but I'd rather not do it in this PR.

Reply:
Is this constructor used for synthesized record property accessors only? If so, and we're not sharing code with other constructors, we should consider simplifying this method. For instance, could `modifiers` and `explicitInterfaceImplementations` be removed?

---
In reply to: [445078938](https://github.com/dotnet/roslyn/pull/45411#discussion_r445078938) [](ancestors = 445078938)